dotnet/Makefile

```diff
@@ -170,14 +171,12 @@ $(foreach platform,$(DOTNET_PLATFORMS),$(eval $(call SupportedTargetPlatforms,$(
 define WorkloadTargets
 Workloads/Microsoft.NET.Sdk.$(1)/WorkloadManifest.json: Makefile $(TOP)/Make.config.inc $(TOP)/.git/HEAD $(TOP)/.git/index Makefile generate-workloadmanifest-json.csharp | Workloads/Microsoft.NET.Sdk.$(1)
 	$$(Q) rm -f $$@.tmp
-	$$(Q_GEN) ./generate-workloadmanifest-json.csharp "$(1)" "$(3)" "$(5)" "$$(DOTNET_$(4)_RUNTIME_IDENTIFIERS)" "$$@.tmp" "$$(DOTNET_WINDOWS_PLATFORMS)"
+	$$(Q_GEN) ./generate-workloadmanifest-json.csharp "$(1)" "$(3)" "$(5)" "$$(DOTNET_$(4)_RUNTIME_IDENTIFIERS)" "$$@.tmp" "$$(DOTNET_WINDOWS_PLATFORMS)" "$(DOTNET_TFM)_$$($(4)_NUGET_OS_VERSION)"
```

Comment:
Looking at the iOS manifest from this pr, the packs section seems to more closely match what .NET 9 is expecting:

```json
"Microsoft.iOS.Sdk.net8.0_17.2",
"Microsoft.iOS.Sdk.net7",
"Microsoft.iOS.Windows.Sdk.Aliased.net8.0_17.2",
"Microsoft.iOS.Windows.Sdk.Aliased.net7",
"Microsoft.iOS.Ref.net8.0_17.2",
"Microsoft.iOS.Runtime.ios-arm64.net8.0_17.2",
"Microsoft.iOS.Runtime.iossimulator-x64.net8.0_17.2",
"Microsoft.iOS.Runtime.iossimulator-arm64.net8.0_17.2",
"Microsoft.iOS.Templates.net8"
```
compared to the workload pack declarations in .NET 9 preview 3:

```json
"Microsoft.iOS.Sdk.net9.0_17.2",
"Microsoft.iOS.Sdk.net8.0_17.0",
"Microsoft.iOS.Sdk.net8.0",
"Microsoft.iOS.Windows.Sdk.Aliased.net9.0_17.2",
"Microsoft.iOS.Windows.Sdk.Aliased.net8.0_17.0",
"Microsoft.iOS.Windows.Sdk.Aliased.net8.0",
"Microsoft.iOS.Ref.net9.0_17.2",
"Microsoft.iOS.Runtime.ios-arm64.net9.0_17.2",
"Microsoft.iOS.Runtime.iossimulator-x64.net9.0_17.2",
"Microsoft.iOS.Runtime.iossimulator-arm64.net9.0_17.2",
"Microsoft.iOS.Templates.net9"
```

The .NET 9 branches would still need to be updated to reference these new pack names however.

@rolfbjarne I'm wondering what the best approach is to sync up these pack names for VS. Renaming here seems higher risk and while I think this may make maintenance easier going forward, maybe it's better to abandon this and continue to special case this branch on the .NET 9 side?

Reply:
Attempt at special-casing in the preview 4 release: https://github.com/xamarin/xamarin-macios/pull/20516